Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -936,23 +936,23 @@ size_t GroupDetectors2::formGroups(API::MatrixWorkspace_const_sptr inputWS,
 
     // Copy over X data from first spectrum, the bin boundaries for all spectra
     // are assumed to be the same here
-    outSpec.dataX() = inputWS->readX(0);
+    outSpec.setSharedX(inputWS->sharedX(0));
 
     // the Y values and errors from spectra being grouped are combined in the
     // output spectrum
-    MantidVec &firstY = outSpec.dataY();
+    auto &firstY = outSpec.mutableY();
     // Keep track of number of detectors required for masking
     size_t nonMaskedSpectra(0);
-    beh->dataX(outIndex)[0] = 0.0;
-    beh->dataE(outIndex)[0] = 0.0;
+    beh->mutableX(outIndex)[0] = 0.0;
+    beh->mutableE(outIndex)[0] = 0.0;
     for (auto originalWI : it->second) {
       // detectors to add to firstSpecNum
       const auto &fromSpectrum = inputWS->getSpectrum(originalWI);
 
       // Add up all the Y spectra and store the result in the first one
-      auto fEit = outSpec.dataE().begin();
-      auto Yit = fromSpectrum.dataY().cbegin();
-      auto Eit = fromSpectrum.dataE().cbegin();
+      auto fEit = outSpec.mutableE().begin();
+      auto Yit = fromSpectrum.y().cbegin();
+      auto Eit = fromSpectrum.e().cbegin();
```

Comment:
There is `operator+` for histograms (you find it in `MantidHistogramData/HistogramMath.h`). The code here assumes that the bin boundaries for all spectra are the same, whereas `operator+` actually checks that. Could you try using `+` here and see if any tests fail?